### PR TITLE
WIP: Upgrade to wxWidgets 3.2

### LINKED ISSUE #31: Update wxWidgets to latest version
## Describe User Story
We are currently using [wxWidgets v2.8.12](https://github.com/wxWidgets/wxWidgets/tree/v2.8.12) which is very old. Update to latest version

Update buckaroo toml in process.

### PR DESCRIPTION
**Pull request by ArenM**
**Testing**: I've built and run this, but I haven't tested it thoroughly yet.

**New Features**: This should make building significantly simpler by having cmake download and build wxWidgets. I think I have static linking working, but `-DwxBUILD_USE_STATIC_RUNTIME=ON` might need to be passed to cmake.

**Background**:  I've been trying to get CedarLogic running on linux, and it turns out this is a large portion of the changes I ended up making. I hope they'll be easier to upstream in a more generally applicable form.

Fixes #31

### CODE REVIEW

**Comment by JoeHCQ1** [comment]
@ArenM this is great! Could you please set the dev-2.4 branch as your destination?

**Comment by JoeHCQ1** [comment]
Awesome! I'll get to trying this out ASAP.

**Comment by ArenM** [comment]
Cool, thanks for the very fast replies!

I did some more testing of this today, and pushed a few new fixes just now. I'm planing on cleaning up my notes about what I tested and include them in the pr description. 

I also got this error after rebasing on dev-2.4, I think it's not related to these changes so I commented out the line to get it to build:
```
`C:\Users\user\CedarLogic\logic\tests\test.cpp(182): error C2146: syntax error: missing ')' before identifier 'and'
C:\Users\user\CedarLogic\logic\tests\test.cpp(182): error C2146: syntax error: missing '(' before identifier 'and'``
```

**Comment by JoeHCQ1** [comment]
@ArenM 

> error after rebase on dev-2.4

I believe that is something wrong with my code. Not really sure why, but that's an existing error in this branch. Haven't tried to address it yet.

I'm going to pull this down now and try building it.

**Comment by JoeHCQ1** [comment]
Confirmed this builds on Windows for the Win32 platform. Just switched it to x64 to see if it still builds (I suspect the old wxWidgets version is the main reason we had to do Win32).

**Comment by JoeHCQ1** [comment]
Probably going to bring this in with fairly minimal testing. The reason why: having to set up and configure wxWidgets 2.8 is killing involvement. It is a hurdle that keeps some from ever getting going, has prevented Linux and Mac contribution, and is certainly not my favorite even on Windows. This enables faster development, which enables faster bug fixing. So it saves more bugs than it brings.

Also, I found the bug behind that compile error. Will fix that in another PR. A `and` needs to be `&&` and we can cut one set of paratheses.

Just waiting to re-run the unit tests locally and see if the x64 build works _after_ I clean the build directory.

**Comment by JoeHCQ1** [comment]
Unit tests pass.

File: src/gui/OscopeCanvas.cpp [hunk 1]
```diff
@@ -212,24 +207,7 @@ void OscopeCanvas::OnPaint(wxPaintEvent& event){
 	SwapBuffers();
 }
 
-void OscopeCanvas::OnSize(wxSizeEvent& event)
```
Comment (JoeHCQ1) [inline src/gui/OscopeCanvas.cpp:215]: Without this function resizing the window on Windows still works.

**Review by JoeHCQ1 — APPROVED** [review]

**Comment by JoeHCQ1** [comment]
@ArenM I'm going to go ahead and merge since I want to keep working off these changes. Feel free to submit more PRs if you find bugs.

Thank you so much for this!!!

**Pull request merged**